### DISCUSSION TITLE
Why incremental analysis does not show only new violations?

### DISCUSSION THREAD

**Discussion by vt89** [pmd/pmd#5002] [Q&A] [answered]
I would understand if we could show only new violations when we use the incremental analysis.
I am using the cache feature and I see the cache is used only during the analysis to speed up the process but the final result contains always all the violations, not only the new one introduced.

Is there any solution to show only the new violations instead of all the ones?

**Comment by jsotuyod** [comment, marked as answer]
No, PMD doesn't support doing differential reports.

PMD will report all violations consistently, the incremental analysis is meant exclusively for performance reasons, but is very much tied to the workflow and pipeline (think branching models, CI/CD pipelines, etc.)… at some point we can probably turn it on by default, but there are some wrinkles to sort out along the way…

Internally, we DO report diffing for regression testing purposes (new PMD version vs previous one), but the way to achieve that is the same you would require: pull the old report, the new one, and diff. Our regression tester is open source, you can see how we do the diffing here https://github.com/pmd/pmd-regression-tester/blob/master/lib/pmdtester/report_diff.rb
